scripts/website/generate-md.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+set -o errexit
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+getName(){
+    name=$(yq e '.spec.crd.spec.names.kind' $1)
+    echo "$name"
+}
+
+getDescription() {
+    description=$(yq e '.metadata.annotations.description' $1)
+    echo "$description"
+}
+
+getTitle() {
+    title=$(yq e '.metadata.annotations.title' $1)
+    echo "$title"
+}
+
+getExamples() {
+    allDetailsExample=""
+    for i in $(ls -d ${1}samples/*/)
+    do
+        constraint=$(cat ${i}/constraint.yaml)
+        constraintExample="<details>\n<summary>constraint</summary>\n\n\`\`\`yaml\n$constraint\n\`\`\`\n\n</details>"
+
+        # find all files that contain the word "allowed" in the filename                
```

Comment:
We should have one SOT for what purpose a given file serves. In this case the `suite.yaml` seems the better candidate to me since it allows for more flexible associations. For example, this `suite.yaml` makes it clear which referential data is associated with which tests:

https://github.com/open-policy-agent/gatekeeper-library/blob/6c3b8a75db8d7b3f0e10ea13d872eef9f3c69d48/library/general/uniqueingresshost/suite.yaml#L1-L25

And does so in a way that would allow for re-use vs. the naming-convention based paradigm.

Another example of edge cases we'll eventually need to support are policies that rely on request metadata, such as CREATE vs UPDATE or requesting user: https://github.com/open-policy-agent/gatekeeper/issues/1819

Right now there are ~2 sources of truth because some of the e2e test stuff still relies on naming conventions, but IMO new stuff should tend toward suite.yaml so we can phase out the use of semantically significant file names.

Reply:
Thanks. `suites.yaml` is more reliable source. I updated script to work off of this.